src/OffsetProgressiveCurve.sol

```diff
@@ -90,17 +90,18 @@ contract OffsetProgressiveCurve is BaseCurve {
     /// @dev or to say that another way:
     /// $$\text{shares} = \sqrt{(s + o)^2 + \frac{2a}{m}} - (s + o)$$
     function previewDeposit(uint256 assets, uint256, /*totalAssets*/ uint256 totalShares)
-        public
```

Comment:
3.20 - Functions in Curve Contracts Could Be external Instead of public

Made all of these `external`